code/modules/mob/living/simple_animal/distortion/nightmare/TimeRipper.dm

```diff
@@ -0,0 +1,203 @@
+	var/stage_threshold = 3000 // enters stage 2 at or below this threshold
+	var/finishing = FALSE
+	var/countering = FALSE //are we //Are you?
+	var/counter_speed = 2 //subtracted from the movedelay when dashing
+	var/counter_ready = FALSE
+	var/damage_taken
+	var/damage_up = 5
+	var/nightmare_mode = FALSE
+	var/frozen_time = 600
+	var/no_dehead = 0
+
+	loot = list(/obj/item/documents/ncorporation, /obj/item/documents/ncorporation)
+
+/mob/living/simple_animal/hostile/distortion/Timeripper/proc/DashCounter() //increases move speed and hits with a powerful attack that knocks back far away
+	playsound(get_turf(src), 'sound/effects/hokma_meltdown.ogg', 75, 0, 3)
+	switch(current_stage)
+		if(1)
+			icon_state = "Ripper"
+		if(2)
+			icon_state = "Ripper2"
+	countering = TRUE
+	counter_ready = FALSE
+	//Speed becomes 4 or 2 and returns to 6 or 4 after 4 seconds.
+	TemporarySpeedChange(-counter_speed, 4 SECONDS)
+	visible_message(span_warning("[src] sprints toward [target] with accelerated speed!"), span_notice("You accelerate self!"), span_notice("You hear footsteps speed up."))
+	addtimer(CALLBACK(src, PROC_REF(DisableCounter)), 4 SECONDS) //disables the counter after 4 seconds
+
+/mob/living/simple_animal/hostile/distortion/Timeripper/proc/DisableCounter() //resets the counter
+	if(countering)
+		countering = FALSE
+		playsound(get_turf(src), 'sound/effects/hokma_meltdown.ogg', 75, 0, 3)
+		SLEEP_CHECK_DEATH(10)
+		icon_state = icon_living
+
+/mob/living/simple_animal/hostile/distortion/Timeripper/OpenFire(target)
+	if(!can_act)
+		return
+
+	if(counter_ready)
+		switch(current_stage)
+			if(1)
+				return DashCounter()
+			if(2)
+				if(prob(80))
+					if(isliving(target))
+						return Timestop(target)
+				return DashCounter()
+		return
+
+
+
+//Phase 2
+/mob/living/simple_animal/hostile/distortion/Timeripper/proc/StageTransition()
+	icon = 'ModularTegustation/Teguicons/64x64.dmi'
+	icon_living = "Ripper2"
+	pixel_x = -16
+	if(!countering && can_act)
+		icon_state = icon_living
+	current_stage = 2
+	counter_threshold = 300
+	playsound(get_turf(src), 'sound/effects/clockcult_gateway_disrupted.ogg', 75, 0, 3)
+
+/mob/living/simple_animal/hostile/distortion/Timeripper/adjustHealth(amount, updating_health = TRUE, forced = FALSE)
+	. = ..()
+	if(. > 0)
+		damage_taken += .
+	if(damage_taken >= counter_threshold && !counter_ready && !countering)
+		counter_ready = TRUE
+		damage_taken = 0
+	if((health <= stage_threshold) && (current_stage == 1))
+		StageTransition()
+
+
+
+/mob/living/simple_animal/hostile/distortion/Timeripper/AttackingTarget(atom/attacked_target)
+	. = ..()
+	if(.)
+		if(finishing)
+			return FALSE
+		if(!istype(attacked_target, /mob/living/carbon/human))
+			return
+		var/mob/living/carbon/human/H = attacked_target
+
+		if(H.health < 0)
+
+			finishing = TRUE
+			switch(current_stage)
+				if(1)
+					icon = 'ModularTegustation/Teguicons/64x64.dmi'
+					icon_state = "Ripper2"
+				if(2)
+					icon_state = "Ripper2"

```

Comment:
This could be compressed a lot, unless thats meant to be two unique sprites
```suggestion
			icon_state = "Ripper2"
			if(current_stage == 1)
				icon = 'ModularTegustation/Teguicons/64x64.dmi'
```